src/sensor/parser-ping.cpp

```diff
@@ -0,0 +1,34 @@
+#include "parser-ping.h"
+
+void PingParserExt::clearBuffer()
+{
+    _parser.reset();
+}
+
+void PingParserExt::parseBuffer(const QByteArray& data)
+{
+    for(int i = 0; i < data.length(); i++) {
+        PingParser::ParseState state = _parser.parseByte(data.at(i));
+        if (state == PingParser::ParseState::NEW_MESSAGE) {
+            parsed++;
+            _rxMessage = _parser.rxMessage;
+            emit newMessage(_rxMessage);
+        } else if (state == PingParser::ParseState::ERROR) {
+            errors++;
+            emit parseError();
+        }
+    }
+}
+
+Parser::ParserState PingParserExt::parseByte(const char byte)
+{
+    uint8_t state = _parser.parseByte(byte);
+    if (state == PingParser::ParseState::NEW_MESSAGE) {
+        _rxMessage = _parser.rxMessage;
+        return Parser::ParserState::NEW_MESSAGE;
+    } else if (state == PingParser::ERROR) {
+        return Parser::ParserState::ERROR;
```

Comment:
Why we are not using the base class enum here ? What is the problem to use PingParser ?

Reply:
> Why we are not using the base class enum here ?

In our project, `Parser::ParseByte` returns the enum type `Parser::ParseState`: https://github.com/bluerobotics/ping-viewer/blob/protocol/src/sensor/parser.h#L19

In ping-cpp, `PingParser::parseByte` returns the enum type `PingParser::ParseState`: https://github.com/bluerobotics/ping-cpp/blob/deployment/src/message/ping-parser.h#L17

The base class Parser has no knowledge of the `PingParser` enum from ping-cpp. The PingParserExt method here does the conversion between the states.

Reply:
`Parser::ParseState::NEW_MESSAGE == PingParser::ParseState::NEW_MESSAGE` is not necessarily true, that's why the code is written like this.